it/src/main/java/org/corfudb/universe/universe/vm/ApplianceManager.java

```diff
@@ -268,4 +266,135 @@ private VirtualMachineCloneSpec createLinuxCustomization(
             this.typeName = typeName;
         }
     }
+
+    @Builder
+    @Slf4j
+    public static class VmManager {
+        @NonNull
+        @Getter
+        private final VmName vmName;
+
+        @NonNull
+        @Getter
+        private final VirtualMachine vm;
+
+        @NonNull
+        private final ServiceInstance serviceInstance;
+
+        @NonNull
+        private final VmUniverseParams universeParams;
+
+        public Result<Void, UniverseException> shutdown() {
+            return Result.of(() -> {
+                try {
+                    vm.shutdownGuest();
+                } catch (RemoteException e) {
+                    throw new UniverseException(e);
+                }
+
+                return null;
+            });
+        }
+
+        public Result<TaskInfo, UniverseException> reset() {
```

Comment:
Is this a guest reset or a cold reboot? They have different side effects, for example the latter can cause half-closed tcp sockets.

Reply:
It a cold reboot. And yes, thank you for the question, I just added `guest reset` the method name is `reboot`.
So:
 reset - cold reset
 reboot - guest reset